packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -762,9 +762,10 @@ export class CommonFrontendContribution implements FrontendApplicationContributi
             }
         });
         commandRegistry.registerCommand(CommonCommands.TOGGLE_BOTTOM_PANEL, {
-            isEnabled: () => this.shell.getWidgets('bottom').length > 0,
             execute: () => {
-                if (this.shell.isExpanded('bottom')) {
+                if (this.shell.getWidgets('bottom').length === 0) {
+                    commandRegistry.executeCommand('terminal:new');
```

Comment:
@shuyaqian I do not believe that `@theia/core` should know anything about terminals.
It is possible that applications will also choose to omit the use of terminals (https://github.com/eclipse-theia/theia/issues/6645).